### PR TITLE
WIP: Add Tor Http Client Factory

### PR DESCRIPTION
**Pull request by Kukks**
This PR adds a Tor Http Client Factory so that we can do HTTP requests over socks5( which we usually use tor's socks5).
Using it for #1321 for sending when possible. Would be great to also use for rates and other external calls if possible.

 Currently have an issue where a 503 is always returned.  Must be something wrong with my tor config or the socks proxy lib Im using.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
The default IHttpClientFactory of dotnet core is a very complicated piece of code that we can't reimplement by ourselves. Instead we should manage to use their extensibility point to support Tor in the default factory.

File: BTCPayServer/BTCPayServer.csproj [hunk 1]
```diff
@@ -35,6 +35,7 @@
     <PackageReference Include="BundlerMinifier.Core" Version="3.2.435" />
     <PackageReference Include="BundlerMinifier.TagHelpers" Version="3.2.435" />
     <PackageReference Include="HtmlSanitizer" Version="4.0.217" />
+    <PackageReference Include="HttpToSocks5Proxy" Version="1.4.0" />
```
Comment (NicolasDorier) [inline BTCPayServer/BTCPayServer.csproj]: No external deps, we have several piece of code to use

https://github.com/NTumbleBit/NTumbleBit/blob/master/NTumbleBit/Tor/SocksMessageHandler.cs

Or better, just use NBitcoin https://github.com/MetacoSA/NBitcoin/blob/master/NBitcoin/Socks/SocksHelper.cs

**Comment by NicolasDorier** [comment]
I searched a bit, and I don't think we have access to underlying socket via the default IHttpClientFactory, so we will probably need to do our own like you do now, but we should limit it only to Payjoin, I don't want to destabilize the rest.

I remember seeing somewhere that in .net5, we will have better access to internal.

**Comment by Kukks** [comment]
Let's revisit this later

**Pull request closed without merging**